htdocs/compta/facture/card.php

```diff
@@ -3730,8 +3730,10 @@ function setRadioForTypeOfIncoice() {
 		print '<input type="hidden" name="amount"   value="'.$objectsrc->total_ht.'">'."\n";
 		print '<input type="hidden" name="total"    value="'.$objectsrc->total_ttc.'">'."\n";
 		print '<input type="hidden" name="tva"      value="'.$objectsrc->total_tva.'">'."\n";
-		print '<input type="hidden" name="origin"   value="'.$objectsrc->element.'">';
-		print '<input type="hidden" name="originid" value="'.$objectsrc->id.'">';
+		// The lines below override the parameters set by GET or POST for origin value and also be present in $origin and $originid variables (it make creation of invoice originating from an external module object impossible because for get an external module class we also need to know the name of the module in addition of the name of the element).
```

Comment:
If we remove this, we will lose them when submitting form and there is an error.
If origin and originid are provided on url, 
$objectsrc->id and $objectsrc->element should have the value of value provided.
So it overwrites yes, but with the same value to guarantee the data is propagated to the post.